### DISCUSSION TITLE
Vim Just Syntax highlighting

### DISCUSSION THREAD

**Discussion by NoahTheDuke** [casey/just#834] [Show and tell]
I was frustrated with the poor overlap of `make` and `just` when using vim, so I whipped up an initial version of a syntax file. I'm still working on it and this is my first time making a syntax file so I'd love for any and all help. Hopefully you can recommend this in the future instead of recommending just reusing `make`'s syntax highlighting.

[vim-just](https://github.com/NoahTheDuke/vim-just)

**Comment by casey** [comment]
Awesome! I agree that the make syntax highlighting is lacking. I tried out `vim-just`, and it definitely looks promising, but there are some bits that it doesn't handle as well as the make syntax highlighting. For example, in the justfile in this repo, it seems to get confused by a `"` inside a recipe body, and highlights everything as a string from then on.

I started writing a test framework for `vim-just`. It uses vim's `TOhtml` command to output the results of syntax highlightning a test file to HTML, which can then be compared to a known-good output. An example of an input:

```
# foo
```

And the corresponding known-good output:

```
<span class="Comment"># foo</span>
```

That way we can add tricky-to-correctly-highlight snippets, and make sure they're highlighted correctly, and also that highlightning doesn't change or regress as `vim-just` is modified.

I'll open a PR soon and you can check it out!

**Reply by NoahTheDuke** [reply]
Thanks so much for trying it out!

`ToHtml` is cool, I didn't know about that. Automated testing will be very helpful.

**Reply by NoahTheDuke** [reply]
I've updated various parts, including `syntax sync fromstart`, which seems to have cleared up the string issue (at least that I could re-create).
